File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        ans = 0
        maxNum, maxCount = max(nums), 0

        l = 0
        for r in range(len(nums)):
            if nums[r] == maxNum: maxCount += 1

            while maxCount == k:
                if nums[l] == maxNum:
                    maxCount -= 1
                l += 1
            
            ans += l
        return ans

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.cs
public class Solution {
    public long CountSubarrays(int[] nums, int k) {
        long ans = 0;
        int maxNum = nums.Max(), maxCount = 0;

        for(int l=0, r=0; r < nums.Length; r++){
            if(nums[r] == maxNum) maxCount++;

            while(maxCount == k)
                if(nums[l++] == maxNum)
                    maxCount--;
            
            ans += l;
        }
        return ans;
    }
}